src/addr-util.c

```diff
@@ -37,24 +37,35 @@ CattaAddress *catta_address_from_sockaddr(const struct sockaddr* sa, CattaAddres
     assert(sa->sa_family == AF_INET || sa->sa_family == AF_INET6);
 
     ret_addr->proto = catta_af_to_proto(sa->sa_family);
-
-    if (sa->sa_family == AF_INET)
-        memcpy(&ret_addr->data.ipv4, &((const struct sockaddr_in*) sa)->sin_addr, sizeof(ret_addr->data.ipv4));
```

Comment:
The variable `sa` always points to a sockaddr_in or a sockaddr_in6, so the alignment is OK. We don't need to do an extra `memcpy()`. That would be weird anyway, `memcpy()` itself doesn't have any alignment requirements, it's unfortunate that Clang doesn't look a bit further. But it can be worked around by casting to `(void *)` like so:
```
memcpy(&ret_addr->data.ipv4, &((const struct sockaddr_in*)(void *) sa)->sin_addr, sizeof(ret_addr->data.ipv4));
```
The same for the other `memcpy()`.